File: sgcmapi/src/main/java/br/ufac/sgcmapi/service/IService.java
package br.ufac.sgcmapi.service;

import java.util.List;

public interface IService<T> {

    public List<T> get();
    public T get(Long id);
    public List<T> get(String termoBusca);
    public T save(T objeto);
    public void delete(Long id);
    
}

File: sgcmapp/src/app/service/i-service.ts
import { Observable } from "rxjs";

export interface IService<T> {

    apiUrl: string;

    get(termoBusca?: string): Observable<T[]>;
    getById(id: number): Observable<T>;
    save(objeto: T): Observable<T>;
    delete(id: number): Observable<void>;

}
